agent/uiserver/uiserver.go

```diff
@@ -117,6 +129,61 @@ func (h *Handler) getState() *reloadableState {
 	return nil
 }
 
+func (h *Handler) serveUIMetricsProviders(resp http.ResponseWriter, req *http.Request) {
+	// Reload config in case it's changed
+	state := h.getState()
```

Comment:
This comment feels rather misleading to me. We're re-reading the state, but not "reloading" config.

We should probably only be calling `getState()` a single time for every request, right? Otherwise we risk it changing values in the middle of a request. This function could accept the config that it uses from state. That way we can have a single `getState()` call in the caller, and the dependencies of this function are more explicit.

Reply:
This is gone thanks to the refactor upstream.